--- prcs-1.3.2/src/rebuild.cc.orig	2002-04-26 15:41:11.000000000 +0200
+++ prcs-1.3.2/src/rebuild.cc	2004-04-05 23:28:11.187829216 +0200
@@ -40,6 +40,11 @@
 #include "syscmd.h"
 #include "checkin.h"
 
+#if __GNUG__ >= 3
+#include <ext/stdio_filebuf.h>
+using namespace __gnu_cxx;
+#endif
+
 struct RebuildCallbackData {
     RepEntry* rep_entry;
     RcsFileTable* file_table;
@@ -995,7 +1000,7 @@
 	buf = new filebuf(seg->fd());
 	buf->seekoff(0,ios::end);
 #else
-        buf = new filebuf(fdopen(dup(seg->fd()), "a+"), ios::out);
+        buf = new stdio_filebuf<char>(fdopen(dup(seg->fd()), "a+"), ios::out);
         buf->pubseekoff(0, ios::end, ios::out);
 #endif
 	os = new ostream(buf);
--- prcs-1.3.2/src/prcserror.cc.orig	2004-04-05 23:19:13.000000000 +0200
+++ prcs-1.3.2/src/prcserror.cc	2004-04-05 23:32:40.220313950 +0200
@@ -51,6 +51,11 @@
 #if defined(__APPLE__)
  stdiobuf stdout_stream(STDOUT_FILENO);
  stdiobuf stderr_stream(STDERR_FILENO);
+#elif __GNUC__ >= 3
+ #include <ext/stdio_filebuf.h>
+ using namespace __gnu_cxx;
+ stdio_filebuf<char> stdout_stream(stdout, ios::out);
+ stdio_filebuf<char> stderr_stream(stderr, ios::out);
 #else
  filebuf stdout_stream(stdout, ios::out);
  filebuf stderr_stream(stderr, ios::out);
